storage-proofs-porep/src/stacked/vanilla/encoding_proof.rs

```diff
@@ -50,15 +49,19 @@ impl<H: Hasher> EncodingProof<H> {
         bytes_into_fr_repr_safe(hasher.finalize().as_ref()).into()
     }
 
-    pub fn verify<G: Hasher>(
+    pub fn verify<D>(
```

Comment:
Note: make `EncodingProof::verify` generic over the TreeD domain `D` rather than hasher `G`; require that TreeR and TreeD use the same field.